notebooks/curation_api/python/src/dataset.py

```diff
@@ -0,0 +1,106 @@
+import boto3
+import logging
+import os
+import requests
+import threading
+from botocore.credentials import RefreshableCredentials
+from botocore.session import get_session
+from datetime import datetime, timezone
+
+from src.utils.logger import get_custom_logger
+
+
+logger = get_custom_logger()
+
+
+def upload_local_datafile(datafile_path: str, collection_uuid: str, identifier: str):
+    """
+    :param datafile_path: the fully qualified path of the datafile to be uploaded
+    :param collection_uuid: the uuid of the Collection to which the resultant Dataset will belong
+    :param identifier: the curator tag or cellxgene Dataset uuid. Must be suffixed with '.h5ad'. See heading
+    of upload_local_datafile.ipynb for details about how to use the identifier to 'create new' vs 'replace existing'
+    :param log_level: the logging level
+    Datasets.
+    :return: None
+    """
+    s3_credentials_path = f"/curation/v1/collections/{collection_uuid}/datasets/s3-upload-credentials"
+    s3_credentials_url = f"{os.getenv('api_url_base')}{s3_credentials_path}"
```

Comment:
add exception / warning and logging for missing api_url_base and access_token

Reply:
alternatively, move `get_headers` and `url_builder` into a common util and leverage them and their existing logging logic

Reply:
> There's a few actions/endpoints not represented here (e.g. checking for dataset status after uploading), is that in scope as a follow-up if this structure is approved?

Yes, precisely!! 🦾